qiita_db/support_files/patches/python_patches/47.py

```diff
@@ -0,0 +1,680 @@
+                sql = """INSERT INTO qiita.logging (time, severity_id, msg)
+                    VALUES (%s, %s, %s)
+                    RETURNING logging_id"""
+                TRN.add(sql, [analysis['timestamp'], 2,
+                              "Unknown error - patch 47"])
+            else:
+                log_id = job_data['log_id']
+
+            # Magic number 4 -> status -> error
+            sql = """UPDATE qiita.processing_job
+                SET processing_job_status_id = 4, logging_id = %s
+                WHERE processing_job_id = %s"""
+            TRN.add(sql, [log_id, job_id])
+
+
+# The new commands that we are going to add generate new artifact types.
+# These new artifact types are going to be added to a different plugin.
+# In interest of time and given that the artifact type system is going to
+# change in the near future, we feel that the easiest way to transfer
+# the current analyses results is by creating 3 different types of
+# artifacts: (1) distance matrix -> which will include the distance matrix,
+# the principal coordinates and the emperor plots; (2) rarefaction
+# curves -> which will include all the files generated by alpha rarefaction
+# and (3) taxonomy summary, which will include all the files generated
+# by summarize_taxa_through_plots.py
+
+# Step 1: Create the new type
+with TRN:
+    # Magic number 2 -> The "artifact definition" software type
+    sql = """INSERT INTO qiita.software
+                (name, version, description, environment_script, start_script,
+                software_type_id)
+             VALUES ('Diversity types', '0.1.0',
+                     'Diversity artifacts type plugin',
+                     'source activate qiita', 'start_diversity_types', 2)
+             RETURNING software_id"""
+    TRN.add(sql)
+    divtype_id = TRN.execute_fetchlast()
+
+    # Step 2: Create the validate and HTML generator commands
+    sql = """INSERT INTO qiita.software_command (software_id, name, description)
+             VALUES (%s, %s, %s)
+             RETURNING command_id"""
+    TRN.add(sql, [divtype_id, 'Validate',
+                  'Validates a new artifact of the given diversity type'])
+    validate_cmd_id = TRN.execute_fetchlast()
+    TRN.add(sql, [divtype_id, 'Generate HTML summary',
+                  'Generates the HTML summary of a given diversity type'])
+    html_summary_cmd_id = TRN.execute_fetchlast()
+
+    # Step 3: Add the parameters for the previous commands
+    sql = """INSERT INTO qiita.command_parameter
+                (command_id, parameter_name, parameter_type, required)
+             VALUES (%s, %s, %s, %s)"""
+    sql_args = [(validate_cmd_id, 'files', 'string', True),
+                (validate_cmd_id, 'artifact_type', 'string', True),
+                (html_summary_cmd_id, 'input_data', 'artifact', True)]
+    TRN.add(sql, sql_args, many=True)
+
+    # Step 4: Add the new artifact types
+    sql = """INSERT INTO qiita.artifact_type (
```

Comment:
This is a general comment. One thing which came up in Krumlov, which almost resulted in @ElDeveloper's death, was that "artifact" has a very negative meaning to biologists. We pinged Greg about it regarding its use in QIIME2 and it is under discussion for possibly changing that the name used.